src/zope/testrunner/runner.py

```diff
@@ -125,6 +127,13 @@ def __init__(self, defaults=None, args=None, found_suites=None,
         self.tests_by_layer_name = {}
 
     def ordered_layers(self):
+        if (self.options.processes > 1 and not self.options.resume_layer):
+            # if we want multiple processes, we need a fake layer as first
+            # to start spreading out layers/tests to subprocesses
+            # but only if this is not in the subprocess
+            yield (name_from_layer(FakeMultiProcessLayer),
+                   FakeMultiProcessLayer, FakeMultiProcessTests_suite())
+
```

Comment:
Ah, I see now that `self.tests_by_layer_name` contains the *filtered* set of test layers to run.

Well, originally it contains the unfiltered set, and then [it gets filtered in-place](https://github.com/zopefoundation/zope.testrunner/blob/master/src/zope/testrunner/filter.py#L31-L53).

(Oh my, now I that I've seen the filtering code feel like I need to make sure unit tests don't get run repeatedly when subprocesses are involved.)